sdk/java-sdk-spring/src/test/scala/kalix/javasdk/impl/JsonMessageCodecSpec.scala

```diff
@@ -76,6 +108,19 @@ class JsonMessageCodecSpec extends AnyWordSpec with Matchers {
       encoded.getTypeUrl shouldBe jsonTypeUrlWith("kalix.javasdk.impl.JsonMessageCodecSpec$SimpleClass")
     }
 
+    "add version number to typeUrl" in {
+      //new codec to avoid collision with SimpleClass
+      val encoded = new JsonMessageCodec().encodeJava(SimpleClassUpdated("abc", 10, 123))
+      encoded.getTypeUrl shouldBe jsonTypeUrlWith("kalix.javasdk.impl.JsonMessageCodecSpec$SimpleClassUpdated#1")
+    }
+
+    "decode with new schema version" in {
+      val encoded = messageCodec.encodeJava(SimpleClass("abc", 10))
+      val decoded =
+        JsonSupport.decodeJson(classOf[SimpleClassUpdated], encoded, Optional.of(new SimpleClassUpdatedMigration))
```

Comment:
I guess we are adding this extra method receiving `Optional<JsonMigration>` because we are using classes that belong to the Protobuf package and we don't have access to `MigrationExtractor.extractMigration` nor the `Migration` annotation. 

IMO, this is another reason to remove this method. We should not increase the public API just because we want to reuse some classes.